### PR TITLE
allow owner to reStake and claim slot with that

### PR DESCRIPTION
**Pull request by 0xAshish**

### CODE REVIEW

File: contracts/staking/stakeManager/StakeManager.sol [hunk 1]
```diff
@@ -191,27 +201,31 @@ contract StakeManager is IStakeManager, StakeManagerStorage {
     ) external onlyWhenUnlocked {
         Auction storage auction = validatorAuction[validatorId];
         Validator storage validator = validators[validatorId];
-        require(msg.sender == auction.user, "Only bidder can confirm");
+        require(
+            msg.sender == auction.user ||
+                getValidatorId(msg.sender) == validatorId,
+            "Only bidder can confirm"
```
Comment (imyourm8) [inline contracts/staking/stakeManager/StakeManager.sol:207]: "Only bidder or validator can confirm" would be more precise.

File: contracts/staking/stakeManager/StakeManager.sol [hunk 2]
```diff
@@ -191,27 +201,31 @@ contract StakeManager is IStakeManager, StakeManagerStorage {
...
+            );
+        }
         // validator is last auctioner
-        if (auction.user == NFTContract.ownerOf(validatorId)) {
-            uint256 refund = validator.amount;
-            require(token.transfer(auction.user, refund));
-            validator.amount = auction.amount;
-
+        if (perceivedStake > auction.amount) {
+            require(token.transfer(auction.user, auction.amount));
```
Comment (imyourm8) [inline contracts/staking/stakeManager/StakeManager.sol:224]: missing revert reason

File: test/units/staking/stakeManager/StakeManager.test.js [hunk 3]
```diff
@@ -1443,133 +1472,64 @@ contract('StakeManager', async function(accounts) {
...
+            from: this.prevValidatorAddr
+          })
+          // restake
+          await this.stakeManager.restake(this.validatorId, restakeAmount, false, {
+            from: this.prevValidatorAddr
+          })
+          this.validator = await this.stakeManager.validators(this.validatorId)
+        })
+
+        testConfirmAuctionBidForOldValidator()
```
Comment (AyushKaul) [inline test/units/staking/stakeManager/StakeManager.test.js:1527]: is this.bidderBalanceBeforeAuction updated for this.prevValidatorAddr? can't see here

Comment (0xAshish) [reply test/units/staking/stakeManager/StakeManager.test.js:1527]: it's in before.

**Comment by imyourm8** [comment]
Fix linter please

File: contracts/staking/stakeManager/StakeManager.sol [hunk 4]
```diff
@@ -191,27 +201,34 @@ contract StakeManager is IStakeManager, StakeManagerStorage {
...
+                ValidatorShare(_contract).activeAmount()
+            );
+        }
         // validator is last auctioner
-        if (auction.user == NFTContract.ownerOf(validatorId)) {
-            uint256 refund = validator.amount;
-            require(token.transfer(auction.user, refund));
-            validator.amount = auction.amount;
-
+        if (perceivedStake > auction.amount) {
```
Comment (atvanguard) [inline contracts/staking/stakeManager/StakeManager.sol:223]: I'd suggest not allowing dethroning with stake == perceivedStake.

Comment (0xAshish) [reply contracts/staking/stakeManager/StakeManager.sol:223]: yeah makes sense.

**Pull request merged**